packages/ui/components/proposal/ProposalStatus.tsx

```diff
@@ -1,5 +1,52 @@
-export interface ProposalStatusProps {}
+import clsx from 'clsx'
+import { ComponentType, SVGProps } from 'react'
 
-export const ProposalStatus = ({}: ProposalStatusProps) => {
-  return <p>placeholder</p>
+import { Approved, Open, Rejected } from '@dao-dao/icons'
+
+export enum EProposalStatus {
+  Open = 'open',
+  Rejected = 'rejected',
+  Approved = 'approved',
+}
+
+export interface ProposalStatusProps {
+  status: `${EProposalStatus}`
+}
+
+export const ProposalStatus = ({ status }: ProposalStatusProps) => {
+  const { Icon, iconClassName, textClassName } = ProposalStatusMap[status]
+
+  return (
+    <div className="flex flex-row gap-2 items-center link-text">
+      <Icon className={clsx('w-[19px] h-[19px]', iconClassName)} />
+      <p className={textClassName}>
+        {status[0].toUpperCase() + status.slice(1)}
```

Comment:
This slicing is a little too magic. What's happening here?

Reply:
Converting to titlecase basically. Capitalizes first letter followed by the rest, and since the enum values are lowercase, the rest is lowercase.

Reply:
I think a `const convertToTitleCase = (str) => ...` could go a long way here.

Reply:
Agreed, this should be a util function, but we can't use those right now because storybook gets mad :( maybe we can add a TODO